### ISSUE TITLE
"Transfer Connection Request" breaks connection statuses

### ISSUE THREAD

**Issue by nlBayside** [SparkDevNetwork/Rock#5162]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Did you perform a search at https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock to see if your bug or enhancement is already reported?
    * Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?
    * Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?

#### A Picture Is worth a Thousand Words
Here is a screenshot of the error that occurs on connection requests that have been transferred to a connection opportunity with different connection statuses.

![Screen Shot 2022-09-07 at 2 22 17 PM](https://user-images.githubusercontent.com/85907597/188983527-c61a925e-ce41-43bc-93dd-809d5224b330.png)

Here is a SQL query showing that the connection statuses are entirely different records.

![Screen Shot 2022-09-07 at 2 43 19 PM](https://user-images.githubusercontent.com/85907597/188988218-866fa6c7-6740-42b5-8e35-6d2e000b1850.png)

### Description
The "Transfer Connection Request" Workflow action does not take into consideration that a connection request has a different connection type than the connection opportunity listed. If the action is executed, then the connection request could potentially have incorrect data.

If the connection opportunity is of the same connection type as the current request's connection opportunity, then this is no problem. Otherwise, the warning pictured above will appear on the Connection Request Detail block.

I think in v14.0 this could be handled easily because the ConnectionRequest model will have a ConnectionTypeId column. This will make an easy comparison between the request and opportunity.

### Steps to Reproduce

1. Create a workflow with a "Transfer Connection Request" action
3. Launch this workflow for a selected Connection Request with a Connection Opportunity that has a different Connection Type.
4. View modified request in the Connection Request Detail block.
5. Click edit on the block.
6. See error

**Expected behavior:**
The workflow action should fail because the request and opportunity are of different connection types.

**Actual behavior:**
The workflow actions is successful and the request now has incorrect connection status for its new type.

### Versions

* **Rock Version:** [12.6 - 13.6]
* **Client Culture Setting:** [en-US]

**Comment by asdf1nit** [comment]
I can confirm this issue but I see it a bit differently. 

The workflow action should let you transfer between types(this is the purpose of the workflow action), but fall back to the new types default status if no "New" status is provided. I have workflows that transfer to different types and in subsequent actions change the status so that there are no errors. 

Stripping the ability to move Connections to a different Type Id would be removing functionality. This could create more admin work as the only way to "move" a connection to a different type is via this action. Without it you would have to create new connections and copy the data over by hand.

**Comment by nlBayside** [comment]
I think my initial problem is that you have to admin work after the transfer to ensure the statuses don't break. It would be great to set a default in the same action! I completely forgot about the "IsDefault" Property on the Connection Status. Sadly, the workflow action does not consider this but it would make sense to include that in it.

I think another thing that would be useful is being able to choose a Connection Status in the workflow, besides the default. So you would combine two actions into one. Thanks for the input! Hopefully this gets resolved eventually

**Comment by nairdo** [comment]
We were not expecting to transfer requests to different opportunities that had different connection types.  We're going to write an error in the workflow if/when that occurs.  We'll also update the documentation to make that more clear.

**Comment by nairdo** [comment]
@nlBayside , @asdf1nit  We discussed this further, and we've decided to add some additional functionality and behavior (described by Jonathan above) to make this action work a bit better.  We are going to add two new optional fields to the workflow action:

- a **Connection Status Id** field (int) (optional) -- if not given **_or_** if the given value is not valid for the destination connection type, then the destination's "Is Default" Connection Status will be used.
- a **Connection State** field (selectable value) (optional) -- if given it will allow the transferred request to have it's State set to this value.
